### DISCUSSION TITLE
nlp_g failed error

### DISCUSSION THREAD

**Discussion by v-upadhyay** [casadi/casadi#3416] [Q&A]
I am trying to implement mpc for a model but it's always showing me "  CasADi - 2023-10-15 14:51:33 WARNING("solver:nlp_g failed: NaN detected for output g, at (row 2, col 1).") [.../casadi/core/oracle_function.cpp:326]  "

**Comment by v-upadhyay** [comment]
(I am using MATLAB 2021a with casadi 3.6.3 on windows 11)

**Reply by sandeep026** [reply]
If you are using Opti( ) class the particular constraint can be examined using

```
opti.debug.x_describe(index)
opti.debug.g_describe(index)
```

Nan usually arises in cases where there is division by zero or square root of a number becomes negative. Hope you have initialized your optimizaiton problem carefully.

**Comment by brunoml5** [comment]
Hi! Take a look at this [FAQ](https://github.com/casadi/casadi/wiki/FAQ%3A-Why-am-I-getting-%22NaN-detected%22in-my-optimization%3F) from the [Github wiki](https://github.com/casadi/casadi/wiki/Onboarding-Guide).
